types/src/user.rs

```diff
@@ -20,3 +21,12 @@ pub struct User {
     /// User-facing name for this contributor/curator
     pub display_name: String,
 }
+
+/// Adds a bookmarked document to the user's list of bookmarks
+#[derive(async_graphql::InputObject)]
+pub struct AddBookmark {
+    /// ID of the document to add bookmark
+    pub document_id: Uuid,
+    /// Whether the document is already bookmarked
+    pub bookmark_bool: bool,
+}
```

Comment:
I think `AddBookmark` is not a great name for something that adds or removes a bookmark. `UpdateBookmark` maybe?